### PR TITLE
Fix bugs in textOffsetsToSelectionState

### PR DESCRIPTION
**Pull request by lukebarnard1**
This just had some thinkos in it. Namely the conditionals were slightly wrong and this lead to negative offset selection state being returned, causing vector-im/riot-web#4792

fixes vector-im/riot-web#4792

### CODE REVIEW

**Comment by t3chguy** [comment]
test flake? (/me has not seen this one before): https://travis-ci.org/matrix-org/matrix-react-sdk/builds/264427882#L918

File: src/RichText.js [hunk 1]
```diff
@@ -204,23 +204,27 @@ export function textOffsetsToSelectionState({start, end}: SelectionRange,
...
         const blockLength = block.getLength();
-        if (start !== -1 && start < blockLength) {
-            selectionState = selectionState.merge({
-                anchorKey: block.getKey(),
-                anchorOffset: start,
-            });
-            start = -1;
-        } else {
-            start -= blockLength + 1; // +1 to account for newline between blocks
+        if (start !== -1) {
```
Comment (ara4n) [inline src/RichText.js]: is -1 a magic constant meaning "unknown" or is it literally off the front of the buffer? some commentary is vital here.

**Comment by ara4n** [comment]
the fix looks plausible, but it's very hard to read the code given the lack of explanation of how the start/end variables are expected to work.

**Comment by ara4n** [comment]
it now looks like i might have more of a chance of understanding how this works in future - thanks ;) lgtm

**Pull request merged**